### PR TITLE
Add stop_orca_context to python exit handler

### LINKED ISSUE #2851: standalone cluster is not shut down if program exit unexpectedly
Maybe using context manager to call `stop_orca_context?`
```python
with OrcaContex() as ctx:
    # user program
```

### PR DESCRIPTION
**Pull request by yangw1234**
Fix #2851

### CODE REVIEW

File: pyzoo/zoo/util/spark.py [hunk 1]
```diff
@@ -205,20 +205,16 @@ def init_spark_standalone(self,
...
-        os.waitpid(stop_master_pro.pid, 0)
+        if env is not None:
+            stop_worker_pro = subprocess.Popen(
+                "{}/sbin/stop-worker.sh".format(env["ZOO_STANDALONE_HOME"]), shell=True, env=env)
+            os.waitpid(stop_worker_pro.pid, 0)
+            stop_master_pro = subprocess.Popen(
+                "{}/sbin/stop-master.sh".format(env["ZOO_STANDALONE_HOME"]), shell=True, env=env)
+            os.waitpid(stop_master_pro.pid, 0)
+        else:
+            # if env is None, then the standalone cluster is not started by analytics zoo
```
Comment (yangw1234) [inline pyzoo/zoo/util/spark.py:216]: Is this correct? @hkvision

Comment (hkvision) [reply pyzoo/zoo/util/spark.py:216]: As I test, seems if the cluster is manually started by the user, this script won't stop it, is it correct?
Previously my implementation is that even if the program exited accidentally, users can open a python shell to stop_orca_context manually.
If this is added to exit handler, then I think stop manually is not needed any longer.

File: pyzoo/zoo/util/spark.py [hunk 2]
```diff
@@ -205,20 +205,16 @@ def init_spark_standalone(self,
...
-            os.waitpid(pro.pid, 0)
-            env = {"SPARK_HOME": pyspark_home,
-                   "ZOO_STANDALONE_HOME": zoo_standalone_home}
-        stop_worker_pro = subprocess.Popen(
-            "{}/sbin/stop-worker.sh".format(env["ZOO_STANDALONE_HOME"]), shell=True, env=env)
-        os.waitpid(stop_worker_pro.pid, 0)
-        stop_master_pro = subprocess.Popen(
-            "{}/sbin/stop-master.sh".format(env["ZOO_STANDALONE_HOME"]), shell=True, env=env)
-        os.waitpid(stop_master_pro.pid, 0)
+        if env is not None:
```
Comment (hkvision) [inline pyzoo/zoo/util/spark.py:208]: just `if env`?

**Review by hkvision — APPROVED** [review]
LGTM

File: pyzoo/zoo/util/spark.py (hunk 2 above)
Comment (yangw1234) [reply pyzoo/zoo/util/spark.py:208]: Though it does not matter in this case, there are some subtle differences between `env` and `env is not None`.
E.g. 
```
env={}
bool(env) # False
env is not None # True
```

**Comment by yangw1234** [comment]
http://172.168.2.101:8080/view/ZOO-PR/job/ZOO-PR-Validation/4197/ jenkin pass

**Pull request merged**